src/open_inwoner/openklant/clients.py

```diff
@@ -1,26 +1,246 @@
+        object_mapping = {}
+        for ocm in object_contact_momenten:
+            assert ocm.contactmoment == contactmoment.url
+            ocm.contactmoment = contactmoment
+            if ocm.object_type == "zaak":
+                object_url = ocm.object
+                # Avoid fetching the same object, if multiple relations wit the same object exist
+                if ocm.object in object_mapping:
+                    ocm.object = object_mapping[object_url]
+                else:
+                    ocm.object = fetch_case_by_url_no_cache(ocm.object)
+                    object_mapping[object_url] = ocm.object
+
+        return object_contact_momenten
+
+    def retrieve_klantcontactmomenten_for_klant(
+        self, klant: Klant
+    ) -> List[KlantContactMoment]:
+        try:
+            response = self.get(
+                "klantcontactmomenten",
+                params={"klant": klant.url},
+            )
+            data = get_json_response(response)
+            all_data = list(pagination_helper(self, data))
+        except (RequestException, ClientError) as e:
+            logger.exception("exception while making request", exc_info=e)
+            return []
+
+        klanten_contact_moments = factory(KlantContactMoment, all_data)
+
+        # resolve linked resources
+        for kcm in klanten_contact_moments:
+            assert kcm.klant == klant.url
+            kcm.klant = klant
+            kcm.contactmoment = self.retrieve_contactmoment(kcm.contactmoment)
+
+        return klanten_contact_moments
+
+    def retrieve_objectcontactmomenten_for_object_type(
+        self, contactmoment: ContactMoment, object_type: str
+    ) -> List[ObjectContactMoment]:
+
+        moments = self.retrieve_objectcontactmomenten_for_contactmoment(contactmoment)
+
+        # eSuite doesn't implement a `object_type` query parameter
+        ret = [moment for moment in moments if moment.object_type == object_type]
+
+        return ret
+
+    def retrieve_objectcontactmoment(
+        self, contactmoment: ContactMoment, object_type: str
+    ) -> Optional[ObjectContactMoment]:
+        ocms = self.retrieve_objectcontactmomenten_for_object_type(
+            contactmoment, object_type
+        )
+        if ocms:
+            return ocms[0]
+
+
 def build_client(type_) -> Optional[APIClient]:
     config = OpenKlantConfig.get_solo()
-    services = {
-        "klanten",
-        "contactmomenten",
+    services_to_client_mapping = {
+        "klanten": KlantenClient,
+        "contactmomenten": ContactmomentenClient,
     }
-    if type_ in services:
+    if client_class := services_to_client_mapping.get(type_):
         service = getattr(config, f"{type_}_service")
         if service:
-            client = _build_client(service, client_factory=JSONParserClient)
+            client = _build_client(service, client_factory=client_class)
             return client
 
     logger.warning(f"no service defined for {type_}")
```

Comment:
`logger.warning("no service defined for {type}", type=type_)` (better to avoid f-string interpolation with logging)

Reply:
`logger.warning("no service defined for %s", type_)` is actually the correct form (see https://docs.python.org/3/howto/logging.html#optimization)

Reply:
I thought `logger.warning("no service defined for {type}", type=type_)` was equivalent to `logger.warning("no service defined for %s", type_)`; perhaps that's not the case.

Reply:
Yeah it's a bit misleading because the doc statement `These newer formatting options _are_ supported` only applies format options, e.g. `%(levelname)s:%(message)s` vs `{levelname}:{message}', not to the actual log messages passed to the logging methods.

Reply:
> I thought `logger.warning("no service defined for {type}", type=type_)` was equivalent to `logger.warning("no service defined for %s", type_)`; perhaps that's not the case.

log methods only allow pos. args (except for `exc_info` and similar). I also think that apart from performance (which can be meaningless is most cases), it allows easier grouping of log messages in tools like Sentry

Reply:
Are we doing log aggregation in tools like Sentry in this project? I don't think the existing logs are setup for that so that would be introducing something new.

Reply:
But it's true that with deferred string interpolation, it would only error at runtime if there's a mismatch in the number of arguments. I think linting rules exist to warn on this issue though

Reply:
> Are we doing log aggregation in tools like Sentry in this project? I don't think the existing logs are setup for that so that would be introducing something new.

We do have Sentry set up for test/acceptance/prod and if `logger.warning(f"no service defined for {type_}")` gets triggered for two different `type_`s, it will create two separate issues in Sentry, whereas `logger.warning(f"no service defined for %s", type_)` will group them together